client/src/test/java/org/opensearch/ml/client/MachineLearningNodeClientTest.java

```diff
@@ -624,14 +624,55 @@ public void deploy() {
         }).when(client).execute(eq(MLDeployModelAction.INSTANCE), any(), any());
 
         ArgumentCaptor<MLDeployModelResponse> argumentCaptor = ArgumentCaptor.forClass(MLDeployModelResponse.class);
-        machineLearningNodeClient.deploy(modelId, DeployModelActionListener);
+        machineLearningNodeClient.deploy(modelId, deployModelActionListener);
 
         verify(client).execute(eq(MLDeployModelAction.INSTANCE), isA(MLDeployModelRequest.class), any());
-        verify(DeployModelActionListener).onResponse(argumentCaptor.capture());
+        verify(deployModelActionListener).onResponse(argumentCaptor.capture());
         assertEquals(taskId, (argumentCaptor.getValue()).getTaskId());
         assertEquals(status, (argumentCaptor.getValue()).getStatus());
     }
 
+    @Test
+    public void createConnector() {
+
+
+        String connectorId = "connectorId";
+
+        doAnswer(invocation -> {
+            ActionListener<MLCreateConnectorResponse> actionListener = invocation.getArgument(2);
+            MLCreateConnectorResponse output = new MLCreateConnectorResponse(connectorId);
+            actionListener.onResponse(output);
+            return null;
+        }).when(client).execute(eq(MLCreateConnectorAction.INSTANCE), any(), any());
+
+        ArgumentCaptor<MLCreateConnectorResponse> argumentCaptor = ArgumentCaptor.forClass(MLCreateConnectorResponse.class);
+
+        Map<String, String> params = Map.ofEntries(Map.entry("endpoint", "endpoint"), Map.entry("temp", "7"));
+        Map<String, String> credentials = Map.ofEntries(Map.entry("key1", "value1"), Map.entry("key2", "value2"));
+        List<String> backendRoles = Arrays.asList("IT", "HR");
+
+        MLCreateConnectorInput mlCreateConnectorInput = MLCreateConnectorInput.builder()
+                .name("test")
+                .description("description")
+                .version("testModelVersion")
+                .protocol("testProtocol")
+                .parameters(params)
+                .credential(credentials)
+                .actions(null)
+                .backendRoles(backendRoles)
+                .addAllBackendRoles(false)
+                .access(AccessMode.from("private"))
+                .dryRun(false)
+                .build();
+
+        machineLearningNodeClient.createConnector(mlCreateConnectorInput, createConnectorActionListener);
+
+        verify(client).execute(eq(MLCreateConnectorAction.INSTANCE), isA(MLCreateConnectorRequest.class), any());
+        verify(createConnectorActionListener).onResponse(argumentCaptor.capture());
+        assertEquals(connectorId, (argumentCaptor.getValue()).getConnectorId());
```

Comment:
May be we can check the status as well?

Reply:
Isn't `MLCreateConnectorResponse` just has [connectorId](https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/transport/connector/MLCreateConnectorResponse.java#L30)?

Reply:
Yeah you are right, Seems like [Status](https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/transport/connector/MLCreateConnectorResponse.java#L20) isn't being used anywhere. Ignore this comment then.